### PR TITLE
Twitter workitem

### PR DESCRIPTION
**Pull request by tsurdilo**
@mswiderski since we could not find the sources of the Twitter workitem that was in older repos, here is impl also using twitter4j.

### CODE REVIEW

File: twitter-workitem/src/main/java/org/jbpm/process/workitem/twitter/TwitterAuth.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import twitter4j.Twitter;
+import twitter4j.conf.ConfigurationBuilder;
+import twitter4j.TwitterFactory;
+
+public class TwitterAuth {
+
+    public Twitter getTwitterService(String consumerKey, String consumerSecret, String accessKey, String accessSecret) {
+        ConfigurationBuilder configBuilder = new ConfigurationBuilder();
+        configBuilder
+                .setDebugEnabled(true)
```
Comment (mswiderski) [inline twitter-workitem/src/main/java/org/jbpm/process/workitem/twitter/TwitterAuth.java]: I would make this configurable

File: twitter-workitem/src/main/java/org/jbpm/process/workitem/twitter/UpdateStatusWorkitemHandler.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        displayName = "TwitterUpdateStatus",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.twitter.UpdateStatusWorkitemHandler()",
+        parameters = {
+                @WidParameter(name = "Message"),
+                @WidParameter(name = "Media")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "org.twitter4j", artifact = "twitter4j-core", version = "4.0.6")
+        })
+public class UpdateStatusWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
```
Comment (mswiderski) [inline twitter-workitem/src/main/java/org/jbpm/process/workitem/twitter/UpdateStatusWorkitemHandler.java]: isn't the name bit confusing? UpdateStatus while it actually sends messages? or maybe I just don't understand twitter

File: twitter-workitem/src/main/java/org/jbpm/process/workitem/twitter/UpdateStatusWorkitemHandler.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                if (statusMedia != null) {
+
+                    File tempMediaFile = File.createTempFile(FilenameUtils.getBaseName(statusMedia.getName()),
+                                                             "." + FilenameUtils.getExtension(statusMedia.getName()));
+
+                    FileOutputStream fos = new FileOutputStream(tempMediaFile);
+                    fos.write(statusMedia.getContent());
+                    fos.close();
+
+                    statusUpdate.setMedia(tempMediaFile);
```
Comment (mswiderski) [inline twitter-workitem/src/main/java/org/jbpm/process/workitem/twitter/UpdateStatusWorkitemHandler.java]: maybe use setMedia(java.lang.String name, java.io.InputStream body)  instead so no temp file is needed

File: twitter-workitem/src/main/java/org/jbpm/process/workitem/twitter/UpdateStatusWorkitemHandler.java (hunk 2 above)
Comment (tsurdilo) [reply twitter-workitem/src/main/java/org/jbpm/process/workitem/twitter/UpdateStatusWorkitemHandler.java]: I think this is fine - code calls twitter.updateStatus. I think you can send direct messages (the "@" someone) and I'll add a workitem for that then as separate thing

**Comment by tsurdilo** [comment]
@mswiderski amended to add workitem handler for sending direct messages (updates test method names so they make more sense)

**Pull request merged**